### PR TITLE
Project Beats: add instantaneous and next beat option for triggers

### PR DESCRIPTION
**Pull request by sanchitmalhotra126**
Adds a dropdown to trigger blocks to select when the trigger should play: immediately, at the next beat, or at the next measure (current behavior).

This change also supports changing dropdown options mid-playback - in that case, the original trigger time of previous triggers will be preserved, but future triggers will use the current value of the dropdown.

Screenshot 2023-05-24 at 3 09 02 PM

## Links

https://codedotorg.atlassian.net/browse/SL-603

## Testing story

Tested locally.

### CODE REVIEW

**Review by mikeharv — COMMENTED** [review]
What will happen we load existing projects that include this block, ie. What’s the default value? If we care about not breaking projects, the default should probably match how it works today. If we don’t care about breaking projects, I’m still curious which feels like the default. Maybe immediately?

**Comment by sanchitmalhotra126** [comment]
> What will happen we load existing projects that include this block, ie. What’s the default value? If we care about not breaking projects, the default should probably match how it works today. If we don’t care about breaking projects, I’m still curious which feels like the default. Maybe immediately?

Good call. We should probably default to the existing behavior which is next measure.

File: apps/i18n/music/en_us.json [hunk 1]
```diff
@@ -38,6 +38,9 @@
   "blockly_fieldRestBeats": "{num} beats",
   "blockly_fieldRestOneMeasure": "1 measure",
   "blockly_fieldRestMeasures": "{num} measures",
+  "blockly_fieldTriggerStartImmediately": "immediately",
+  "blockly_fieldTriggerStartNextBeat": "at next beat",
+  "blockly_fieldTriggerStartNextMeasure": "at next measure",
```
Comment (breville) [inline apps/i18n/music/en_us.json:43]: Mildly curious how this these fragments (and the containing block) will localize...

**Comment by breville** [comment]
Just curious: is it deliberate that the default is "immediately" for new uses of this block?

**Review by breville — APPROVED** [review]
Works nicely, and opens up some fun new creativity!

**Comment by sanchitmalhotra126** [comment]
> Just curious: is it deliberate that the default is "immediately" for new uses of this block?

I'll update so it defaults to 'next measure', to preserve behavior

**Pull request merged**